shm/inc/Player.hpp

```diff
@@ -2,21 +2,26 @@
 
 #include "shm/inc/Cargo.hpp"
 #include "shm/inc/Ship.hpp"
+#include "shm/inc/Delegate.hpp"
 
 #include <algorithm>
 #include <memory>
 #include <numeric>
 
-class Player {
+class Player : public Delegate {
 public:
-    Player(std::unique_ptr<Ship>, size_t, size_t);
+    Player(std::unique_ptr<Ship> ship, size_t money, size_t availableSpace);
+    ~Player() override;
```

Comment:
No need to declare as there is nothing to do (no definition in cpp). Destructor doesn't have to be pure virtual in Delegate header.